src/audio/dai.c

```diff
@@ -473,6 +473,14 @@ static int dai_params(struct comp_dev *dev,
 			comp_err(dev, "dai_params(): failed to alloc dma buffer");
 			return -ENOMEM;
 		}
+
+		/*
+		 * dma_buffer frame_fmt should be overwritten with value
+		 * received in sof_ipc_comp_config, because value from params
+		 * represents data format for component connected with dai
+		 */
+		buffer_set_params(dd->dma_buffer, params, true);
+		dd->dma_buffer->stream.frame_fmt = dconfig->frame_fmt;
```

Comment:
This is unconditional override while dai_comp_get_hw_params() tests != 0. Is this asymmetry intentional? If so, please also explain inside the comment.

Reply:
removed condition from dai_comp_get_hw_params()